lib/client.ts

```diff
@@ -81,6 +81,33 @@ class GitHubAPIClient {
       const content = Buffer.from(contentResponse.data.content, 'base64').toString('utf-8')
       const titleMatch = content.match(/title:\s*(.+)/)
       const dateMatch = content.match(/date:\s*(.+)/)
+      
+      // Extract external discussions from frontmatter
+      const externalDiscussions: ExternalDiscussion[] = []
+      const frontmatterMatch = content.match(/^---\n([\s\S]*?)\n---/)
+      if (frontmatterMatch) {
+        const frontmatter = frontmatterMatch[1]
+        const discussionsMatch = frontmatter.match(/external_discussions:\s*\n([\s\S]*?)(?=\n\w|\n$)/)
+        if (discussionsMatch) {
+          const discussionsText = discussionsMatch[1]
+          const discussionLines = discussionsText.split('\n').filter(line => line.trim())
+          
+          let currentDiscussion: Partial<ExternalDiscussion> = {}
+          for (const line of discussionLines) {
+            if (line.includes('platform:')) {
+              if (currentDiscussion.platform && currentDiscussion.url) {
+                externalDiscussions.push(currentDiscussion as ExternalDiscussion)
+              }
+              currentDiscussion = { platform: line.split(':')[1].trim() as ExternalDiscussion['platform'] }
+            } else if (line.includes('url:')) {
+              currentDiscussion.url = line.split('url:')[1].trim()
+            }
+          }
+          if (currentDiscussion.platform && currentDiscussion.url) {
+            externalDiscussions.push(currentDiscussion as ExternalDiscussion)
+          }
+        }
+      }
 
```

Comment:
Since we have a lot codes reading information from markdown file, we better to create a module to deal with markdown and parse the proper information, and ensure unit test cases cover.